File: tests/test_interpreter.py
from eaplanner.interpreter import AbsoluteScheduleInterpreter

from eaplanner.entities.schedule import Schedule
from eaplanner.entities.assignment import Assignment
import numpy as np


def test_absolute_interpret():
    # arrange
    assignments = [
        Assignment(),
        Assignment(),
        Assignment(),
    ]
    schedule = Schedule(assignments)
    interpreter = AbsoluteScheduleInterpreter(schedule)
    chromosome = np.array([0, 1, 9, 1, 3, 1])

    # act
    interpreter.interpret(chromosome)

    # assert
    assert assignments[0].start == 0
    assert assignments[0].duration == 1
    assert assignments[1].start == 9
    assert assignments[1].duration == 1
    assert assignments[2].start == 3
    assert assignments[2].duration == 1
    assert schedule.get_total_makespan() == 10
    assert schedule.get_total_penalty() == 0

File: eaplanner-rs/src/test/test_interpreter.rs
use crate::{
    entities::{assignment::Assignment, schedule::Schedule},
    interpreter::{AbsoluteScheduleInterpreter, Interpreter},
};

#[test]
fn test_absolute_interpret() {
    // arrange
    let assignments = vec![
        Assignment::new(None, 10),
        Assignment::new(None, 10),
        Assignment::new(None, 10),
    ];
    let mut schedule = Schedule::new();
    schedule.add_assignments(assignments);
    let mut interpreter = AbsoluteScheduleInterpreter::new(schedule);
    let chromosome = vec![0.0, 1.0, 9.0, 1.0, 3.0, 1.0];

    // act
    interpreter.interpret(&chromosome);

    // assert
    assert_eq!(interpreter.schedule.assignments[0].start(), 0);
    assert_eq!(interpreter.schedule.assignments[0].duration(), 1);
    assert_eq!(interpreter.schedule.assignments[1].start(), 9);
    assert_eq!(interpreter.schedule.assignments[1].duration(), 1);
    assert_eq!(interpreter.schedule.assignments[2].start(), 3);
    assert_eq!(interpreter.schedule.assignments[2].duration(), 1);
    assert_eq!(interpreter.schedule.get_total_makespan(), 10);
    assert_eq!(interpreter.schedule.get_total_penalty(), 0);
}
